gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,588 @@
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET") && path.startsWith("/v1beta1/projects/")) {
+        response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+      } else if (requestMethod.equals("GET")) {
+        response = list(parseListOptions(exchange.getRequestURI().getQuery()));
+      } else if (requestMethod.equals("PUT")) {
+        String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+        response = replace(jsonFactory.fromString(requestBody, Project.class));
+      }
+      if (response == null) {
+        throw new UnsupportedOperationException("Request not recognized.");
+      }
+      exchange.getResponseHeaders().set("Content-type", "application/json; charset=UTF-8");
+      exchange.sendResponseHeaders(response.code(), response.body().length());
+      OutputStream outputStream = exchange.getResponseBody();
+      outputStream.write(response.body().getBytes());
+      outputStream.close();
+    }
+  }
+
+  private static String decodeContent(Headers headers, InputStream inputStream) throws IOException {
+    List<String> contentEncoding = headers.get("Content-encoding");
+    byte[] bytes;
+    if (contentEncoding != null && contentEncoding.size() > 0
+        && contentEncoding.get(0).contains("gzip")) {
+      bytes = ByteStreams.toByteArray(new GZIPInputStream(inputStream));
+      log.fine("Content-encoding is in gzip format. Decoded successfully.");
+    } else {
+      bytes = ByteStreams.toByteArray(inputStream);
+    }
+    return new String(bytes, StandardCharsets.UTF_8);
+  }
+
+  private static String projectIdFromURI(String path) {
+    String[] pathSplit = path.split("/");
+    if (pathSplit.length < 4) {
+      throw new IllegalArgumentException("This path doesn't have a project ID");
+    }
+    return path.split("/")[3].split(":")[0];
+  }
+
+  private static String[] parseFields(String query) {
+    return query != null ? query.split("=")[1].split(",") : null;
+  }
+
+  private static Map<String, Object> parseListOptions(String query) {
+    Map<String, Object> options = new HashMap<>();
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        switch (argEntry[0]) {
+          case "fields":
+            options.put("fields", argEntry[1].split(","));
+            break;
+          case "filter":
+            options.put("filter", argEntry[1].split(" "));
+            break;
+          case "pageToken":
+            // support pageToken when Cloud Resource Manager supports this (#421)
+            break;
+          case "pageSize":
+            // support pageSize when Cloud Resource Manager supports this (#421)
+            break;
+        }
+      }
+    }
+    return options;
+  }
+
+  private static final boolean isValidProject(Project project) {
```

Comment:
Nit, probably more useful would have been to return the error string as the message for the error response.